server/data/strapiApiClient.ts

```diff
@@ -126,7 +126,7 @@ export default class StrapiApiClient {
     teamId?: number
     teamSlug?: string
     withEnvironments?: boolean
-  }): Promise<SingleResponse<Team>> {
+  }): Promise<SingleResponse<Team> | ListResponse<Team>> {
```

Comment:
This pattern is manky - 
If it's search by slug it returns a list with one it, if it's search by id it returns a single response.

The service layer always then has to determine which type of response and potentially unwrap the array to get the item out - this should all be handled at this layer so the service layer doesn't need to know that strapi doesn't natively support search by slug